diff --git a/src/utils.ts b/src/utils.ts
index 66166d2c8fd901246d3f25bbb9347b2ab833afa5..dfeb51979639412cf14f5f3d5f8fec9f96938b8a 100644
--- a/src/utils.ts
+++ b/src/utils.ts
@@ -119,13 +119,14 @@ function getEntities(tweet: TweetBase): Entity[] {
   const result: EntityWithType[] = [
     { indices: tweet.display_text_range, type: "text" },
   ];
-
-  addEntities(result, "hashtag", tweet.entities.hashtags);
-  addEntities(result, "mention", tweet.entities.user_mentions);
-  addEntities(result, "url", tweet.entities.urls);
-  addEntities(result, "symbol", tweet.entities.symbols);
-  if (tweet.entities.media) {
-    addEntities(result, "media", tweet.entities.media);
+  const entities = tweet.entities ?? {};
+
+  addEntities(result, "hashtag", entities.hashtags);
+  addEntities(result, "mention", entities.user_mentions);
+  addEntities(result, "url", entities.urls);
+  addEntities(result, "symbol", entities.symbols);
+  if (entities.media) {
+    addEntities(result, "media", entities.media);
   }
   fixRange(tweet, result);
 
@@ -156,8 +157,10 @@ function getEntities(tweet: TweetBase): Entity[] {
 function addEntities(
   result: EntityWithType[],
   type: EntityWithType["type"],
-  entities: TweetEntity[],
+  entities: TweetEntity[] | undefined,
 ) {
+  if (!entities) return;
+
   for (const entity of entities) {
     for (const [i, item] of result.entries()) {
       if (
@@ -193,11 +196,9 @@ function addEntities(
  * Array.from is unicode aware, unlike string.slice()
  */
 function fixRange(tweet: TweetBase, entities: EntityWithType[]) {
-  if (
-    tweet.entities.media &&
-    tweet.entities.media[0].indices[0] < tweet.display_text_range[1]
-  ) {
-    tweet.display_text_range[1] = tweet.entities.media[0].indices[0];
+  const media = tweet.entities?.media;
+  if (media && media[0].indices[0] < tweet.display_text_range[1]) {
+    tweet.display_text_range[1] = media[0].indices[0];
   }
   const lastEntity = entities.at(-1);
   if (lastEntity && lastEntity.indices[1] > tweet.display_text_range[1]) {
